packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -1266,8 +1414,23 @@ Future<TimeOfDay> showTimePicker({
 }) async {
   assert(context != null);
   assert(initialTime != null);
+
+  // Announce that the time picker is being shown and currently selected time.
+  final MediaQueryData media = MediaQuery.of(context);
+  final MaterialLocalizations localizations = MaterialLocalizations.of(context);
+  _announceToAccessibility(localizations.formatTimeOfDay(initialTime, alwaysUse24HourFormat: media.alwaysUse24HourFormat));
+
   return await showDialog<TimeOfDay>(
     context: context,
     child: new _TimePickerDialog(initialTime: initialTime),
   );
 }
+
+void _announceToAccessibility(String message) {
+  SystemChannels.accessibility.send(<String, dynamic>{
+    'type': 'announce',
+    'data': <String, dynamic>{
+      'message': message,
+    },
+  });
```

Comment:
i thought we had an API for this.

cc @amirh, who I think was working on that.

Reply:
Yes we have [SemanticsService.announce](https://master-docs-flutter-io.firebaseapp.com/flutter/semantics/SemanticsService/announce.html)

Reply:
Ooh! Thanks for adding this. Done.